### PR TITLE
Update submodules

### PR DESCRIPTION
**Pull request by waahm7**
*Description of changes:*
- update to latest submodules
- [Here](https://github.com/actions/runner-images/issues/6840) it is announced that dotnet 5.0 will be removed on 29th Jan. Add a setup dotnet action for 64bit Windows.
- For 32 bit windows, we disabled the tests at [link](https://github.com/awslabs/aws-crt-dotnet/blob/2e6c8bfc102a746c8adc13d4330dd0888f541c31/tests/tests.csproj#L14). Disable it in builder.json as well. We can enable it after setup dotnet action supports installing 32 bit .net. There is a [PR](https://github.com/actions/setup-dotnet/pull/338) in progress for that.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -94,6 +94,10 @@ jobs:
       matrix:
         arch: [x64]
     steps:
+    - name: Setup dotnet
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: '5.0.x'
     - uses: ilammy/msvc-dev-cmd@v1
```
Comment (graebm) [inline .github/workflows/ci.yml:101]: not your fault, but

`ilammy/msvc-dev-cmd` <-- in my experience, this github action has no effect

it ended up in all our ci.yml files when we were first setting up our repos and copy/pasting everything

later we discovered it had no effect, but we've never made a concerted effort to fix this across our 20+ repos :(

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -110,6 +114,10 @@ jobs:
       matrix:
         arch: [x86, x64]
     steps:
```
Comment (graebm) [inline .github/workflows/ci.yml:116]: not your fault, but this "windows-vc14" run isn't doing what it claims.

vc14 means version 14.0 of the compiler, aka toolset 140, which came with Visual Studio 2015, but you can still use those older toolset versions with later versions of Visual Studio

but it's actually just using the default compiler on this machine, which is Visual Studio 2019's version 14.29

Here's the [line in the logs](https://github.com/awslabs/aws-crt-dotnet/actions/runs/4008641738/jobs/6883066390#step:4:174) where you see what it's actually using.
>   -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x86/cl.exe

and then you cross-reference that against the [Wikipedia page that details what the hell those version numbers mean](https://en.wikipedia.org/wiki/Microsoft_Visual_C%2B%2B#Internal_version_numbering). And you see that 14.29 means it's the latest build Visual Studio 2019 compiler

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -94,6 +94,10 @@ jobs:
       matrix:
         arch: [x64]
     steps:
+    - name: Setup dotnet
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: '5.0.x'
```
Comment (graebm) [inline .github/workflows/ci.yml:100]: it's not clear why this CI step is using `setup-dotnet` but none of the other CI steps are using it. Add a comment saying why you're using it

File: .github/workflows/ci.yml (hunk 1 above)
Comment (waahm7) [reply .github/workflows/ci.yml:101]: https://tenor.com/en-GB/view/coding-programming-pink-panther-when-you-delete-a-block-of-code-that-you-though-was-useless-gif-17338075

File: .github/workflows/ci.yml (hunk 3 above)
Comment (waahm7) [reply .github/workflows/ci.yml]: Thanks, fixed.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (waahm7) [reply .github/workflows/ci.yml:117]: Added a TODO for now.

**Review by graebm — APPROVED** [review]

**Pull request merged**